### PR TITLE
Use track start time (ISO 8601) and track name when exporting

### PR DESCRIPTION
**Pull request by rgmf**
**Describe the pull request**
When export all tracks the name of the files follow this rule:
`<track start time in ISO 8601 compact format>-<trackName>.<extension>`

For example, if you export to KMZ a track called "Passo dello Stelvio by bike" started at "2015-07-29T08:21:33Z" results in a file exported name like this:
`20150729082133-Passo dello Stelvio by bike.kmz`

Every time a export all process is launched exports all tracks without duplication.

The rule for not duplication is that files with the same date time and same extension are the same, so this process will overwrites it.

In this way users can export the same track with several extensions. So you can get:
`20150729082133-Passo dello Stelvio by bike.kmz`
`20150729082133-Passo dello Stelvio by bike.gpx`
`20150729082133-Passo dello Stelvio by bike.kml`

**Important**: this is not compatible with old OpenTracks installations. If a user has his tracks exported in the old way and this user exports all his tracks with the new way then this user will have track files duplicated in the storage. That's why this PR is a Draft PR.

**Tests**
Tests has been developed too.

**Link to the the issue**
https://github.com/OpenTracksApp/OpenTracks/issues/159

**License agreement**
By opening this pull request, you are providing your contribution under the _Apache License 2.0_ (see [LICENSE.md](LICENSE.md)).

### CODE REVIEW

File: src/androidTest/java/de/dennisguse/opentracks/io/file/exporter/ExportAsyncTaskTest.java [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+import de.dennisguse.opentracks.content.data.TestDataUtil;
+import de.dennisguse.opentracks.content.data.Track;
+import de.dennisguse.opentracks.content.data.TrackPoint;
+import de.dennisguse.opentracks.content.provider.ContentProviderUtils;
+import de.dennisguse.opentracks.io.file.TrackFileFormat;
+import de.dennisguse.opentracks.stats.TrackStatistics;
+import de.dennisguse.opentracks.util.FileUtils;
+import de.dennisguse.opentracks.util.TrackNameUtils;
+
+public class ExportAsyncTaskTest {
```
Comment (dennisguse) [inline src/androidTest/java/de/dennisguse/opentracks/io/file/exporter/ExportAsyncTaskTest.java:35]: RunWith annotation is missing.
Is it not necessary?

Add a comment that this track deletes all data.
See #143 (also add a comment there).

File: src/androidTest/java/de/dennisguse/opentracks/io/file/exporter/ExportAsyncTaskTest.java [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+        exportDirectoryFile.mkdir();
+        exportDirectory = DocumentFile.fromFile(exportDirectoryFile);
+
+        Assert.assertTrue(exportDirectoryFile.isDirectory());
+    }
+
+    @After
+    public void tearDown() {
+        for (int i = 0; i < trackIdList.length; i++) {
+            contentProviderUtils.deleteAllTracks(context);
```
Comment (dennisguse) [inline src/androidTest/java/de/dennisguse/opentracks/io/file/exporter/ExportAsyncTaskTest.java:78]: This should not happen in the for loop.
At least it looks weird.

File: src/androidTest/java/de/dennisguse/opentracks/io/file/exporter/ExportAsyncTaskTest.java [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+            contentProviderUtils.deleteTrack(context, trackIdList[i]);
+        }
+
+        FileUtils.deleteDirectoryRecurse(exportDirectoryFile);
+    }
+
+    private String addOneDay(String dateTimeStr) throws ParseException {
```
Comment (dennisguse) [inline src/androidTest/java/de/dennisguse/opentracks/io/file/exporter/ExportAsyncTaskTest.java:85]: static

File: src/main/java/de/dennisguse/opentracks/io/file/exporter/ExportAsyncTask.java [hunk 4]
```diff
@@ -158,6 +152,42 @@ public void onProgressUpdate(int number, int max) {
...
 
+    /**
+     * Build the DocumentFile where track will be exported.
+     * The DocumentFile can be a new one or an existing one (overwriting).
+     * The file's name will be a prefix follow by the track name follow by a suffix.
+     *
+     * @param track the Track object needed to build the name.
+     * @return the DocumentFile where track will be exported.
+     */
+    private DocumentFile buildFile(Track track) {
```
Comment (dennisguse) [inline src/main/java/de/dennisguse/opentracks/io/file/exporter/ExportAsyncTask.java:163]: Move to utils class (static method)?

File: src/androidTest/java/de/dennisguse/opentracks/io/file/exporter/ExportAsyncTaskTest.java (hunk 1 above)
Comment (rgmf) [reply src/androidTest/java/de/dennisguse/opentracks/io/file/exporter/ExportAsyncTaskTest.java:35]: > RunWith annotation is missing.
> Is it not necessary?

I sincerely don't know, I don't notice any difference but I annotated it. I'm learning Java Testing on the way (I'm mainly Python/Django developer) :sweat_smile:


> Add a comment that this track deletes all data.
> See #143 (also add a comment there).

Why does testing use the Application Context? Shouldn't testing use Mock Context or Instrumentation Context? Maybe is this the issue? I'm really newbie in these issues so sorry if I'm saying a stupid thing :relaxed:

Anyway, commented done.

File: src/androidTest/java/de/dennisguse/opentracks/io/file/exporter/ExportAsyncTaskTest.java (hunk 2 above)
Comment (rgmf) [reply src/androidTest/java/de/dennisguse/opentracks/io/file/exporter/ExportAsyncTaskTest.java:78]: Ops, sorry. Fixed ;)

**Comment by rgmf** [comment]
@dennisguse this PR can be closed. It doesn't sense anymore.

**Pull request closed without merging**